routers/repo/compare.go

```diff
@@ -139,28 +159,93 @@ func ParseCompareInfo(ctx *context.Context) (*models.User, *models.Repository, *
 	ctx.Data["BaseIsBranch"] = baseIsBranch
 	ctx.Data["BaseIsTag"] = baseIsTag
 
-	// Check if current user has fork of repository or in the same repository.
-	headRepo, has := models.HasForkedRepo(headUser.ID, baseRepo.ID)
-	if !has && !isSameRepo {
+	// Now we have the repository that represents the base
+
+	// The current base and head repositories and branches may not
+	// actually be the intended branches that the user wants to
+	// create a pull-request from - but also determining the head
+	// repo is difficult.
+
+	// We will want therefore to offer a few repositories to set as
+	// our base and head
+
+	// 1. First if the baseRepo is a fork get the "ForkedRepo" it was
+	// forked from
+	var forkedRepo *models.Repository
+	if baseRepo.IsFork {
+		err = baseRepo.GetBaseRepo()
```

Comment:
```suggestion
		err = baseRepo.LoadBaseRepo()
```
Maybe this name is more reasonable for this function.

Reply:
That's a refactor that I don't particularly want to get into in this because if I go anywhere near that code it's all getting renamed.

`Base` is too overloaded. It needs to become `Root` or `ForkBase`.

Reply:
Frankly none of this is reasonable.